xugrid/ugrid/ugridbase.py

```diff
@@ -1032,6 +1045,50 @@ def sel_points(
             selection = selection.where(condition, other=fill_value)
         return selection
 
+    def locate_nearest_node(self, points: FloatArray, max_distance: float = np.inf):
+        """
+        Find which grid node is nearest for a collection of points.
+
+        Parameters
+        ----------
+        points: ndarray of floats with shape ``(n_point, 2)``
+        max_distance: optional, float
+
+        Returns
+        -------
+        indices: ndarray of integers with shape ``(n_point,)``
+            Missing indices are indicated with -1.
+        """
+        _, indices = self.node_kdtree.query(
+            points, distance_upper_bound=max_distance, workers=-1
+        )
+        # The scipy KDTree returns missing indices (e.g. out of max_distance) with n.
+        # We use -1 for consistency.
+        indices[indices == self.n_node] = -1
```

Comment:
Maybe good to add this to the docstring to avoid confusion for people used to scipy's KDTree?